src/find-nargo.ts

```diff
@@ -1,6 +1,15 @@
+import os from 'os';
+import path from 'path';
+import fs from 'fs';
 import which from 'which';
+import { NargoNotFoundError } from './noir';
+import { MarkdownString } from 'vscode';
 
-const nargoBinaries = ['nargo'];
+// List of possible nargo binaries to find on Path
+// We prioritize 'aztec-nargo' as more specialised case
```

Comment:
Related to the above: We can't always prioritise aztec-nargo as it's essentially the same as vanilla nargo but just released at different commits. This then leads to the situation where using the LSP's codelenses could compile to a different program compared to running `nargo compile` in the CLI (or not compile at all!)

These are going to cause very tricky to debug issues for users so we should try to make sure we always use the correct version of nargo (although ideally `aztec-nargo` would not exist).

Reply:
difference between compiler tools are going to be possible not matter if we consider aztec-nargo in here...?